app/models/nearby_facility.rb

```diff
@@ -82,6 +86,26 @@ def empty_resource_set?(response_body)
       response_body['resourceSets'].size.zero? || response_body['resourceSets'][0]['estimatedTotal'].zero?
     end
 
+    def waypoint(params)
+      type = location_type(params)
+      values = REQUIRED_PARAMS[type].map { |k| params[k].to_s }
+      case type
+      when :address
+        return values.join(' ')
+      when :lat_lng
+        return values.join(',')
+      else
+        return ''
+      end
+    end
```

Comment:
It seems like the API interface has burrowed it's way down into the model layer. I could see a cleaner separation of concerns between the controller and the model if the model provided explicit methods for each type of query. Extending that idea, since each method would have it's own interface, the parameter requirements could be enforced by using kwargs.

Reply:
@dvogel hopefully you can point me in the right direction.  I'm having some trouble using kwargs.. I refactored the NearbyFacility code to have two methods now (one for lat/lng, other for address) and I am using metaprogramming to call the appropriate method in NearbyController but it doesn't work when I converted the two NearbyFacility query methods to use kwargs.

I know the first problem I ran into was that it complained that the controllers params isn't a hash so I used

`resource = query_method.call(params.permit!.to_hash)`

but I seem to now run into a different issue and not sure what it would be.